invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/MemberSearchBar.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * This file is part of Invenio.
+ * Copyright (C) 2022 CERN.
+ *
+ * Invenio is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { List, Dropdown } from "semantic-ui-react";
+import { i18next } from "@translations/invenio_communities/i18next";
+import { Image, withCancel } from "react-invenio-forms";
+import _isEmpty from "lodash/isEmpty";
+
+export class MembersSearchBar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isFetching: false,
+      error: false,
+    };
+  }
+
+  serializeUsersForDropdown = (users) => {
+    return users.map((person) => {
+      const name = person.profile.full_name
+        ? person.profile.full_name
+        : person.id;
+      return {
+        text: person.profile.full_name,
+        value: person.id,
+        key: person.id,
+        content: (
+          <List divided relaxed>
+            <List.Item>
+              <Image size="mini" src={person.links.avatar} avatar />
+              <List.Content>
+                <List.Header as="a">{name}</List.Header>
+                <List.Description>
+                  {person.profile.affiliations}
+                </List.Description>
+              </List.Content>
+            </List.Item>
+          </List>
+        ),
+      };
+    });
+  };
+
+  serializeGroupsForDropdown = (groups) => {
+    return groups.map((group) => {
+      return {
+        text: group.name,
+        value: group.id,
+        key: group.id,
+        content: (
+          <List divided relaxed>
+            <List.Item>
+              <Image size="mini" src={group.links.avatar} avatar />
+              <List.Content>
+                <List.Header as="a">{group.name}</List.Header>
+              </List.Content>
+            </List.Item>
+          </List>
+        ),
+      };
+    });
+  };
+
+  optionsGenerator = (suggestions) => {
+    const { searchType } = this.props;
+    if (searchType === "user") {
+      return this.serializeUsersForDropdown(suggestions);
+    } else if (searchType === "group") {
+      return this.serializeGroupsForDropdown(suggestions);
+    }
+  };
+
+  onChange = (event, data) => {
+    const { handleChange, selectedMembers, suggestions, searchType } =
+      this.props;
+    if (
+      !selectedMembers.some(
```

Comment:
functions like these inside conditions block can be hard to understand with a glance, i suggest to place inside a variable which describes what the condition should be 

```js
const condition = () => {}
if(condition)
```